Client/Frontend/Browser/BrowserViewController/BrowserViewController+Menu.swift

```diff
@@ -184,16 +184,7 @@ extension BrowserViewController {
   }
 
   private func presentWallet() {
-    let privateMode = PrivateBrowsingManager.shared.isPrivateBrowsing
-    guard let walletStore = WalletStore.from(privateMode: privateMode) else {
-      log.error("Failed to load wallet. One or more services were unavailable")
-      return
-    }
-    self.onPendingRequestUpdatedCancellable = walletStore.onPendingRequestUpdated
-      .sink { [weak self] _ in
-        self?.updateURLBarWalletButton()
-      }
-
+    guard let walletStore = self.walletStore ?? newWalletStore() else { return }
```

Comment:
I moved the log.error in the `newWalletStore()` function below which includes a guard on `WalletStore.from(privateMode:)` to avoid duplicate loc. I can remove from there and add to the two spots we call this `newWalletStore()` function instead if that helps?